projects/Weblate/service.nix

```
@@ -0,0 +1,408 @@
+    with open("${cfg.smtp.passwordFile}") as f:
+      EMAIL_HOST_PASSWORD = f.read().rstrip("\n")
+
+    CELERY_TASK_ALWAYS_EAGER = False
+    CELERY_BROKER_URL = "redis+socket://${config.services.redis.servers.weblate.unixSocket}"
+    CELERY_RESULT_BACKEND = CELERY_BROKER_URL
+
+    ${cfg.extraConfig}
+  '';
+  settings_py = pkgs.runCommand "weblate_settings.py" {} ''
+    mkdir -p $out
+    cat ${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/settings_example.py > $out/settings.py
+    cat >> $out/settings.py <<EOF${weblateConfig}EOF
+  '';
+  uwsgiConfig.uwsgi = {
+    type = "normal";
+    plugins = ["python3"];
+    master = true;
+    socket = "/run/weblate.socket";
+    die-on-idle = true;
+    die-on-term = true;
+    idle = 600;
+    manage-script-name = true;
+    cheap = true;
+    chmod-socket = "770";
+    chown-socket = "weblate:weblate";
+    uid = "weblate";
+    gid = "weblate";
+    wsgi-file = "${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/wsgi.py";
+    pyhome = pkgs.weblate.dependencyEnv;
+
+    # Some more recommendations by upstream:
+    # https://docs.weblate.org/en/latest/admin/install.html#sample-configuration-for-nginx-and-uwsgi
+    buffer-size = 8192;
+    reload-on-rss = 250;
+    workers = 8;
+    enable-threads = true;
+    close-on-exec = true;
+    umask = "0022";
+    ignore-sigpipe = true;
+    ignore-write-errors = true;
+    disable-write-exception = true;
+  };
+  environment = {
+    PYTHONPATH = "${settings_py}";
+    DJANGO_SETTINGS_MODULE = "settings";
+    # This could be moved into the makeWrapperArgs of the package,
+    # but then uwsgi would be missing out on these variables.
+    GI_TYPELIB_PATH = lib.makeSearchPathOutput "out" "lib/girepository-1.0" (with pkgs; [
+      pango
+      harfbuzz
+      librsvg
+      gdk-pixbuf
+      glib
+    ]);
+  };
+  weblate-env =
+    pkgs.writeShellScriptBin
+    "weblate-env"
+    ((pkgs.lib.concatStrings (pkgs.lib.mapAttrsToList (n: v: "export ${n}=${v}\n") environment))
+      + ''
+        eval -- "\$@"
+      '');
+  weblatePath = with pkgs; [
+    gitSVN
+
+    #optional
+    git-review
+    tesseract
+    licensee
+    mercurial
+  ];
+in {
+  options = {
+    services.weblate = {
+      enable = lib.mkEnableOption "Weblate service";
+
+      localDomain = lib.mkOption {
+        description = "The domain serving your Weblate instance.";
+        example = "weblate.example.org";
+        type = lib.types.str;
+      };
+
+      djangoSecretKeyFile = lib.mkOption {
+        description = ''
+          Location of the Django secret key.
+
+          This should be a string, not a nix path, since nix paths are copied into the world-readable nix store.
+        '';
+        type = lib.types.path;
+      };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = ''
+          Text to append to <filename>settings.py</filename> Weblate config file.
```

```suggestion
          Text to append to `settings.py` Weblate config file.
```

`<filename>` is not supported as child of `<td>`, generated in overview.

---

Done!